provider/datagen/src/source.rs

```diff
@@ -34,6 +35,10 @@ impl Default for SourceData {
                 AbstractFs::new(PathBuf::from(std::env!("CARGO_MANIFEST_DIR")).join("data"))
                     .expect("valid dir"),
             )),
+            segmenter_lstm_paths: Arc::new(JsonCache::new(
+                AbstractFs::new(PathBuf::from(std::env!("CARGO_MANIFEST_DIR")).join("data"))
```

Comment:
Please add some structure to the `data` directory. At least have one directory per cache here.

Reply:
The problem is that some of the files are JSON and some of them are TOML. I think @makotokato's solution here is valid and reasonable. It may just be a naming thing; rather than `segmenter_paths` and `segmenter_lstm_paths`, maybe it should be `datagen_data_toml_paths` and `datagen_data_json_paths`

Reply:
To clarify, I think this code is good, and it should be `segmenter_paths` and `segmenter_lstm_paths`. I'm saying the `data` directory shouldn't be flat but we should put the LSTM files in `data/lstm`.